Code Review

JayKay24 opened the pull request.

JayKay24 committed:
Update README file

JayKay24 committed:
feat(user-registration): Register users into the application.

This commits adds a feature that enables user registration and login into the application.

[Delivers #159619666]

JayKay24 committed:
chore(Home page): Add a landing page and navigation

This commit adds a landing page and navigation to various endpoints in the application.

JayKay24 committed:
fix(styling): Fix navigation styling

This commit modifys and fixes navigation styling to
match the overall theme of the application.

JayKay24 committed:
feat(vehicle-cards): Display vehicle information

This commit adds cards used to display vehicle
information.]

JayKay24 committed:
fix(restful-actions): Include restful actions in controllers

This commit modifies controller actions into more restful ones

JayKay24 committed:
feat(rides): Create Rides

This commit adds the functionality needed to
create a ride offer.

JayKay24 committed:
feat(subscription): Subscribe to a Ride Offer

This commit adds the functionality needed to
subscribe to a ride offer.

JayKay24 committed:
feat(unsubscribe): Unsubscribe from a ride

This commit adds the functionality needed to unsubscribe
from a ride offer.

[Delivers #159685826]

JayKay24 committed:
feat(cancel-ride): Cancel a Ride Offer

This commit adds the functionality required to cancel
a ride offer.

[Delivers #159685954]

JayKay24 committed:
feat(update-ride): Update a ride offer

This commit adds the functionality required
to update a ride offer.

[Delivers #159686468]

JayKay24 committed:
Merge pull request #1 from JayKay24/cancel-ride-offer-159685954

Core functionality of the application

JayKay24 committed:
Merge pull request #2 from JayKay24/update-a-ride-offer-159686468

Update Ride offer

JayKay24 committed:
chore(Boostrapping Rails BDD): Bootstrapping RSpec BDD

This commit bootstraps cucumber, rspec, factory-bot and
shoulda-matchers with the application.

JayKay24 committed:
chore(authentication): Authentication acceptance test

This commit adds an acceptance test for the authentication
feature.

JayKay24 committed:
chore(travis): Configure travis

This commit adds configuration for travis.yml for
coveralls.

morriskimani commented on app/controllers/subscriptions_controller.rb:
```diff
@@ -0,0 +1,46 @@
+      ride_id: params[:id]
+    )
+    subscription.destroy
+    ride.update(remaining_seats: ride.remaining_seats + 1)
```
I suggest that you have this in an `after_destroy` callback in your subscription model. One advantage of doing it this way is that both operations (destroying the subscription and incrementing remaining_seats) are wrapped inside a transaction. Therefore, if any one of them fails for some reason, changes will be rolled back.

morriskimani commented on app/controllers/vehicles_controller.rb:
```diff
@@ -0,0 +1,61 @@
+  before_action :authenticate_user, only: %i[index show new create update destroy]
+
+  def index
+    @vehicles = @current_user.vehicles.all
```
I don't think you have to call `.all` here. `@current_user.vehicles` returns all vehicles belonging to the current user

morriskimani commented on app/controllers/vehicles_controller.rb:
```diff
@@ -0,0 +1,61 @@
+  end
+
+  def create
+    if vehicle_params[:no_of_seats].to_i <= 0
```
This should be a validation on your model

morriskimani commented on app/controllers/vehicles_controller.rb:
```diff
@@ -0,0 +1,61 @@
+  end
+
+  def update
+    if vehicle_params[:no_of_seats].to_i <= 0
```
same here. Let's do this validation at the model level

morriskimani reviewed (changes requested):
@JayKay24 good work so far.
I've left some few comments for you to look at when you start refactoring.

morriskimani commented on app/controllers/rides_controller.rb:
```diff
@@ -0,0 +1,63 @@
+  def update
+    @ride = Ride.find(ride_params[:id])
+    @ride.update(ride_params)
+    if @ride.save
```
```
if @ride.update(ride_params) ...
```

JayKay24 committed:
chore(refactoring): Refactor codebase to be more
efficient.

[Fixes #160221333]

JayKay24 committed:
fix(tests): Clean up tests

This commit adds code elegance to tests by reducing the
lines of repeated code and using built in methods in
favor of writing custom methods.

[Fixes #160221333]

JayKay24 committed:
Merge pull request #6 from JayKay24/ch-refactoring-tests-160221333

Refactoring tests.

JayKay24 committed:
Merge pull request #4 from JayKay24/ch-refactoring-160221333

Refactoring Tests and Models
